app/views/agencies/show.html.erb

```diff
@@ -5,11 +5,14 @@
 
 <span  class="page-header articleTitle">
   <h1 class="page-header articleTitle"><%= @agency.name %></h1>
-  <h4><%= @agency.street_address %></h4>
-  <h4><%= @agency.city %>, <%= State.find(@agency.state_id).name %> <%= @agency.zipcode %></h4>
```

Comment:
Yeah, it's pretty simple, just replace
` <%= State.find(@agency.state_id).name %> `
with a scope perhaps named
` <%= State.from_agency(@agency).name %>